cairo1-run/src/cairo_run.rs

```diff
@@ -885,6 +886,19 @@ fn check_only_array_felt_input_type(
         false
     }
 }
+
+// Returns true if the generic id corresponds to an implicit argument (aka a builtin, gas, or system type)
+fn is_implicit_generic_id(generic_ty: &GenericTypeId) -> bool {
+    generic_ty == &SegmentArenaType::ID
+        || generic_ty == &GasBuiltinType::ID
+        || generic_ty == &BitwiseType::ID
+        || generic_ty == &EcOpType::ID
+        || generic_ty == &PedersenType::ID
+        || generic_ty == &PoseidonType::ID
+        || generic_ty == &RangeCheckType::ID
+        || generic_ty == &SegmentArenaType::ID
+        || generic_ty == &SystemType::ID
```

Comment:
Nit: a shorter way to express the same:
```suggestion
    [
        SegmentArenaType::ID,
        GasBuiltinType::ID,
        BitwiseType::ID,
        EcOpType::ID,
        PedersenType::ID,
        PoseidonType::ID,
        RangeCheckType::ID,
        SegmentArenaType::ID,
        SystemType::ID,
    ].contains(generic_ty)
```

Reply:
better than clippy